packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,356 @@
+    function lockGold(uint256 value) external onlyBeneficiary returns (bool) {
+
+      // the beneficiary may not lock more than the vesting has currently released
+      uint256 unreleasedAmount = _getReleasableAmount(block.timestamp);
+      require(unreleasedAmount >= value, "Gold Amount to lock must not be less that the currently releasable amount");
+
+      bool success;
+      (success,) = address(getLockedGold()).lock.gas(gasleft()).value(msg.value)();
+      require(success);
+      emit VestingGoldLocked(msg.value, block.timestamp);
+      return success;
+    }
+
+    /**
+     * @notice A wrapper func for the unlock gold method function
+     * @param value the value to gold to be unlocked for the vesting instance
+     * @return True if the transaction succeeds.
+     * @dev To be called only by the beneficiary of the vesting
+     */
+    function unlockGold(uint256 value) external onlyBeneficiary returns (bool) {
+      bool success;
+      (success,) = address(getLockedGold()).unlock.gas(gasleft()).value(msg.value)(value);
+      require(success);
+      emit VestingGoldUnlocked(msg.value, block.timestamp);
+      return success;
+    }
+
+    /**
+     * @notice A wrapper func for the relock locked gold method function
+     * @param index the index of the pending locked gold withdrawal
+     * @return True if the transaction succeeds.
+     * @dev To be called only by the beneficiary of the vesting.
+     */
+    function relockLockedGold(uint256 index) external onlyBeneficiary returns (bool) {
+      bool success;
+      (success,) = address(getLockedGold()).relock.gas(gasleft())(index);
+      require(success);
+      emit VestingGoldRelocked(index, block.timestamp);
+      return success;
+    }
+
+    /**
+     * @notice A wrapper func for the withdraw locked gold method function
+     * @param index the index of the pending locked gold withdrawal
+     * @return True if the transaction succeeds.
+     * @dev To be called only by the beneficiary of the vesting. The amount shall be withdrawn back by the vesting instance
+     */
+    function withdrawLockedGold(uint256 index) external onlyBeneficiary returns (bool) {
+      bool success;
+      (success,) = address(getLockedGold()).withdraw.gas(gasleft())(index);
+      require(success);
+      emit VestingGoldWithdrawn(index, block.timestamp);
+      return success;
+    }
+
+    /**
+     * @notice A wrapper func for the authorize vote signer account method
+     * @param v The recovery id of the incoming ECDSA signature.
+     * @param r Output value r of the ECDSA signature.
+     * @param s Output value s of the ECDSA signature.
+     * @return True if the transaction succeeds.
+     * @dev To be called only by the beneficiary of the vesting. The v,r and s signature should be a signed message by the beneficiary being the vesting contract instance address
+     */
+    function authorizeVoteSigner(uint8 v, bytes32 r, bytes32 s) external onlyBeneficiary returns (bool) {
```

Comment:
`address signer` should be a parameter, which we pass to `authorizeVoteSigner`. Ditto for the other `authorize*` functions

Reply:
done